beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -208,6 +214,45 @@ func (s *Service) processPendingBlocks(ctx context.Context) error {
 	return s.sendBatchRootRequest(ctx, parentRoots, randGen)
 }
 
+func (s *Service) requestPendingBlobs(ctx context.Context, b interfaces.ReadOnlyBeaconBlock, br []byte) error {
+	// Block before deneb has no blob.
+	if b.Version() < version.Deneb {
+		return nil
+	}
+	c, err := b.Body().BlobKzgCommitments()
+	if err != nil {
+		return err
+	}
+	// No op if the block has no blob commitments.
+	if len(c) == 0 {
+		return nil
+	}
+	// Choose the best peer to request blob sidecars.
+	_, peers := s.cfg.p2p.Peers().BestFinalized(maxPeerRequest, s.cfg.chain.FinalizedCheckpt().Epoch)
+	if len(peers) == 0 {
+		return errors.New("no peers to request blob sidecars")
+	}
+	// Build request for blob sidecars.
+	bid := make([]*eth.BlobIdentifier, len(c))
+	for i := range c {
+		bid[i] = &eth.BlobIdentifier{Index: uint64(i), BlockRoot: br}
+	}
+
+	ctxByte, err := ContextByteVersionsForValRoot(s.cfg.chain.GenesisValidatorsRoot())
+	if err != nil {
+		return err
+	}
+	req := p2ptypes.BlobSidecarsByRootReq(bid)
+
+	// Send request to a random peer.
+	i := rand.NewGenerator().Int() % len(peers)
+	blobSidecars, err := SendBlobSidecarByRoot(ctx, s.cfg.clock, s.cfg.p2p, peers[i], ctxByte, &req)
```

Comment:
Since blob availbility is not guaranteed, how likely is that a random peer will have a blob ? If there is a good chance they don't we might want to have a round robin where we ask a random selection of peers for it.

Reply:
the view should be cohesive, a block can't be processed without a blob and here we are syncing to the head